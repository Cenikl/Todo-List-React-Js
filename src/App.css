.App{
  text-align: center;
}
.work {
  display: flex;
  flex-direction: row;
  margin-top: 10px;
}
@media screen and (max-width: 550px){
  .work{
    flex-direction: column;
  }
  .done{
    margin-top: 5px;
  }
} 
.todo {
  flex: 50%;
  text-align: center;
  border: 1px solid black;
  border-radius: 5px;
}
.done {
  flex: 50%;
  text-align: center;
  border: 1px solid black;
  border-radius: 5px;
}

.add{
  display: flex;
  flex-direction: row;
  justify-content: center;
}
.titre{
  text-align: center;
  border: 1px solid black;
  padding: 5px;
  border-radius: 5px;
}
.desc{
  text-align: center;
  border: 1px solid black;
  padding: 5px;
  border-radius: 5px;
}

.listing {
  display: flex;
  flex-direction: row;
}
.listing p {
  flex: 25%;
}
.listing input{
  flex: 10%;
  margin: 45px;
}

