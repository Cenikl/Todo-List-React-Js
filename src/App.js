import React, { useState } from 'react';
import './App.css';
import { Todo } from './component/todo';
import { Done } from './component/done';

/*function App() {
 const [number,setNumber] = useState(0);
  return (
    <div className="App">
      <p>{number}</p>
        <button onClick={() => setNumber(number + 1)}>Add</button>
      <button onClick={() => setNumber(0)}>Reset</button>
    </div>
  );
}*/

/*class App extends React.Component {
  constructor(){
    super();
    this.state = {number : 0};
  }

  addNumber = () => {
    this.setState({number : this.state.number + 1})
  }
  resetNumber = () => {
    this.setState({number : 0})
  }

  render(){
    return (
      <div className="App">
        <p >{this.state.number}</p>
        <button onClick={this.addNumber}>Add</button>
        <button onClick={this.resetNumber}>Reset</button>
      </div>
    )
  }
}*/
function App(){
  const [newTitle,setNewTitle] = useState("");
  const [items,setItems] = useState([]);
  const [finished,setFinished] = useState([]);

  return(
    <div className="App">
      <div className="work">
      <Todo 
      items={items}
      setItems={setItems}
      finished={finished}
      setFinished={setFinished}
      newTitle={newTitle}
      setNewTitle={setNewTitle}
      />
      <Done
      finished={finished}
      />
      </div>
    </div>
  );
}

export default App;
