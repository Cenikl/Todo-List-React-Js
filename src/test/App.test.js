import { render, screen, cleanup} from '@testing-library/react';
import App from '../App';
import renderer from 'react-test-renderer';

afterEach(() =>{
    cleanup();
})

it("renders without crashing",()=>{
   render(<App/>);
   const appElement = screen.getByTestId("app-1");
   expect(appElement).toBeInTheDocument();
   expect(appElement).toHaveTextContent('Todo List App');
   expect(appElement).toHaveTextContent('Done');
   
})

it("matches snapshot",()=>{
    const tree = renderer.create(<App/>).toJSON();
    expect(tree).toMatchSnapshot();
})

