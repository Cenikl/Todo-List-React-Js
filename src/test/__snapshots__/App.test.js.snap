// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  data-testid="app-1"
>
  <div
    className="work"
  >
    <div
      className="todo"
      data-testid="todo-1"
    >
      <h1>
        Todo List App
      </h1>
      <hr />
      <div
        className="inputing"
      >
        <input
          className="titre"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Put a Title here..."
          type="text"
          value=""
        />
        <br />
      </div>
      <ul />
    </div>
    <div
      className="done"
      data-testid="done-1"
    >
      <h1>
        Done
      </h1>
      <hr />
      <ul />
    </div>
  </div>
</div>
`;
